papertty.py

```diff
@@ -537,6 +582,73 @@ def sigusr1_handler(sig, frame):
                 oldbuff = ''
                 flags['scrub_requested'] = False
+
+            if flags['show_menu']:
+                flags['show_menu'] = False
+                print()
+                print('Rendering paused. Enter')
+                print('    (f) to change font,')
+                print('    (s) to change spacing,')
+                if ptty.is_truetype:
+                    print('    (h) to change font size,')
+                print('    (x) to exit,')
+                print('    anything else to continue.')
+                print('Use --font {} --size {} --spacing {} for the current setting.'.format(ptty.fontfile, ptty.fontsize, ptty.spacing))
+
+                ch = sys.stdin.readline().strip()
+                if ch == 'x':
+                    if not noclear:
+                        ptty.showtext(oldbuff, fill=ptty.white, **textargs)
+                    sys.exit(0)
+                elif ch == 'f':
+                    print('Current font: {}'.format(ptty.fontfile))
+                    font_name = click.prompt('Enter new font (leave empty to abort)')
+                    if font_name:
+                        ptty.spacing = spacing
+                        ptty.font = ptty.load_font(font_name, keep_if_not_found=True)
+                        if autofit:
+                            max_dim = ptty.fit(portrait)
+                            print("Automatic resize of TTY to {} rows, {} columns".format(max_dim[1], max_dim[0]))
+                            ptty.set_tty_size(ptty.ttydev(vcsa), max_dim[1], max_dim[0])
+                        oldbuff = None
+                    else:
+                        print('Font not changed')
+                elif ch == 's':
+                    print('Current spacing: {}'.format(ptty.spacing))
+                    new_spacing = None
+                    try:
+                        new_spacing = int(click.prompt('Enter new spacing (leave empty to abort)'))
+                    except:
+                        pass
+                    if new_spacing or new_spacing == 0:
+                        ptty.spacing = new_spacing
+                        ptty.recalculate_font()
+                        if autofit:
+                            max_dim = ptty.fit(portrait)
+                            print("Automatic resize of TTY to {} rows, {} columns".format(max_dim[1], max_dim[0]))
+                            ptty.set_tty_size(ptty.ttydev(vcsa), max_dim[1], max_dim[0])
+                        oldbuff = None
+                    else:
+                        print('Spacing not changed')
+                elif ch == 'h' and ptty.is_truetype:
+                    print('Current font size: {}'.format(ptty.fontsize))
+                    new_fontsize = None
+                    try:
+                        new_fontsize = int(click.prompt('Enter new font size (leave empty to abort)'))
+                    except:
+                        pass
+                    if new_fontsize:
```

Comment:
As with the spacing setting, this could also be written as:

```python
new_fontsize = click.prompt('Enter new font size (leave empty to abort)', default='empty', type=int, show_default=False)
if new_fontsize != 'empty':

```